c/logging.c

```diff
@@ -647,8 +652,115 @@ int logGetLevel(LoggingContext *context, uint64 compID){
   return component ? component->currentDetailLevel : ZOWE_LOG_NA;
 }
 
-void zowelog(LoggingContext *context, uint64 compID, int level, char *formatString, ...){
+// creates a formatted timestamp to be used in zowe log
+static void getLogTimestamp(char *date) {
+   int64 stck = 0;      
+   char timestamp[14]; // unformated time
+   uint32 out_secs = 0;
+   uint32 out_micros = 0;
+
+   getSTCK(&stck);
+   int64 unixTime = stckToUnix(stck); // gets unixTime
+   stckToUnixSecondsAndMicros(stck, &out_secs, &out_micros); // calculates microseconds
+   out_micros = out_micros/1000;
+   unixToTimestamp((uint64) unixTime, timestamp);
+   
+   // format: 2019-03-19 11:23:57.776
+   snprintf(date,24, "%.4s-%.2s-%.2s %.2s:%.2s:%.2s.%.3d"
+      ,timestamp, timestamp+4,timestamp+6,timestamp+8,timestamp+10,timestamp+12,(unsigned int) out_micros);
+}
 
+static void getLevel(int level, char *logLevel) {
+  
+  switch(level) {
+    case 0: snprintf(logLevel,7,"SEVERE");
+            break;
+    case 1: snprintf(logLevel,8,"WARN");
+            break;
+    case 2: snprintf(logLevel,5,"INFO"); 
+            break;
+    case 3: snprintf(logLevel,5,"DEBUG");
+            break;
+    case 4: snprintf(logLevel,6,"DEBUG");
+            break;
+    case 5: snprintf(logLevel,7,"TRACE");
+            break;
+  }
+}
+
+static void getLocationData(char *path, int line, char **locationInfo, uint64 compID, LoggingComponent *component) {
+  
+  char prefix[128]; 
+  char suffix[128]; 
+  
+  unsigned int id = compID & 0xFFFFFFFF;
+  if(compID >= LOG_PROD_COMMON && compID < LOG_PROD_ZIS) {
+    snprintf(prefix,5,"_zcc");
+    switch (id) { // most of these don't actally log, but are defined in logging.h
+      case 0x10000: snprintf(suffix,6,"alloc");
+                    break;
+      case 0x20000: snprintf(suffix,6,"utils");
+                    break;
+      case 0x30000: snprintf(suffix,12,"collections");    
+                    break;
+      case 0x40000: snprintf(suffix,14,"serialization");
+                    break;
+      case 0x50000: snprintf(suffix,9,"zlparser");
+                    break;
+      case 0x60000: snprintf(suffix,11,"zlcompiler");
+                    break;
+      case 0x70000: snprintf(suffix,10,"zlruntime"); 
+                    break;
+      case 0x80000: snprintf(suffix,8,"stcbase");
+                    break;
+      case 0x90000: snprintf(suffix,11,"httpserver");
+                    break;
+      case 0xA0000: snprintf(suffix,10,"discovery");
+                    break;
+      case 0xB0000: snprintf(suffix,12,"dataservice");
```

Comment:
dataservice is here, yet unixfile is in zss, and neither restfile nor restdataservice are here, since I made them afterward. May want to revisit.